infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java

```diff
@@ -172,9 +179,14 @@ private InfrastructureProvisionResult doProvision(InfrastructureConfig infrastru
                 logger.info(StringUtil.concatStrings("Stack configuration created for name ", stackName));
             }
             logger.info(StringUtil.concatStrings("Waiting for stack : ", stackName));
-            Waiter<DescribeStacksRequest> describeStacksRequestWaiter = new AmazonCloudFormationWaiters(cloudFormation)
-                    .stackCreateComplete();
-            describeStacksRequestWaiter.run(new WaiterParameters<>(new DescribeStacksRequest()));
+            StackCreationValidator stackCreationValidator = new StackCreationValidator();
+            try {
+                stackCreationValidator.waitForStack(
+                        TIMEOUT, TIMEOUT_UNIT, POLL_INTERVAL, POLL_UNIT, stackName, cloudFormation);
```

Comment:
I think cloudformation and stackName need to be the first two arguments.

Reply:
Fixed in 84fae83